### PR TITLE
RavenDB-5625: /admin/debug/cpu&proc/stats

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            var djv = new DynamicJsonValue();
+            foreach (var prop in typeof(Process).GetProperties())
+            {
+                try
+                {
+                    var value = prop.GetValue(proc);
+
+                    if (value != null)
+                    {
+                        if ( value.GetType() == typeof(ProcessThreadCollection))
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs]: `value is ProcessThreadCollection ptc`

File: src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            };
+        }
+
+        private DynamicJsonValue ProcStatsInternal()
+        {
+            var proc = Process.GetCurrentProcess();
+
+            var dja = new DynamicJsonArray();
+            var djv = new DynamicJsonValue();
+            foreach (var prop in typeof(Process).GetProperties())
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs]: Why are you doing this dynamically?

Comment (aviviadi) [reply src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs]: The /debug/cpu/stats is a summarized (not dynamically generated) end point.  I added this /debug/proc/stats dynamically as it contains a lot of info which can be useful when debugging, and is also changing from one to another dotnet release.   I though reflection suits here.  Should I manually write each property?

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs]: Yes, better that way. Feel free to generate the code, but it is much better to read it in this manner.

Comment (aviviadi) [reply src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+        private static void AddValue<T>(DynamicJsonValue djv, string key, Func<T> func)
+        {
+            try
+            {
+                var result = func.Invoke();
+
+                if (result == null)
+                {
+                    djv[key] = "null";
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs:208]: Why not just set to null directly?

File: src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+                }
+
+                djv[key] = result;
+
+            }
+            catch (Exception ex)
+            {
+                if (ex.InnerException != null)
+                {
+                    if (ex.InnerException.Message.Contains("Process must exit before requested information can be determined") == false)
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs:219]: Can you tell which ones would need that and skip them actively?

Comment (aviviadi) [reply src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs:219]: https://github.com/ravendb/ravendb/pull/3763

File: src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs (hunk 3 above)
Comment (aviviadi) [reply src/Raven.Server/Documents/Handlers/Debugging/ProcStatsHandler.cs:208]: https://github.com/ravendb/ravendb/pull/3763

**Pull request merged**